python-selenium/src/bandcamp/web/element.py

```diff
@@ -0,0 +1,84 @@
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+
+from bandcamp.web.base import Track, WebComponent
+from bandcamp.web.locators import DiscoverPageLocator, TrackLocator
+
+
+class TrackElement(WebComponent, TrackLocator):
+    """Model a playable track in Bandcamp's Discover section."""
+
+    def play(self) -> None:
+        """Play the track."""
+        if not self.is_playing:
+            self._get_play_button().click()
+
+    def pause(self) -> None:
+        """Pause the track."""
+        if self.is_playing:
+            self._get_play_button().click()
+
+    @property
+    def is_playing(self) -> bool:
+        return "Pause" in self._get_play_button().get_attribute("aria-label")
+
+    def _get_track_info(self) -> Track:
+        """Create a representation of the track's relevant information."""
+        full_url = self._parent.find_element(*self.URL).get_attribute("href")
+        # Cut off the referrer query parameter
+        clean_url = full_url.split("?")[0] if full_url else ""
+        # Some tracks don't have a genre
+        try:
+            genre = self._parent.find_element(*self.GENRE).text
+        except NoSuchElementException:
+            genre = ""
+        return Track(
+            album=self._parent.find_element(*self.ALBUM).text,
+            artist=self._parent.find_element(*self.ARTIST).text,
+            genre=genre,
+            url=clean_url,
+        )
+
+    def _get_play_button(self):
+        return self._parent.find_element(*self.PLAY_BUTTON)
+
+
+class DiscoverTrackList(WebComponent, DiscoverPageLocator, TrackLocator):
```

Comment:
@martin-martin I believe I mentioned this before, but I'm not a big fan of abusing inheritance for code reuse like here, where you inherit the constans. Class mixins are primarily about combining _behaviors_. When you only have data attributes, it's clearer to use composition and delegation instead.

For example, instead of this:

```python
view_more_button = self._driver.find_element(*self.PAGINATION_BUTTON)
```

...you'd have that:

```python
class DiscoverTrackList(WebComponent):
    # ...
    view_more_button = self._driver.find_element(*DiscoverPageLocator.PAGINATION_BUTTON)
```

While more verbose, this version tells me exactly which web component the locator belongs to, so I don't have to hunt for it in the source code.